jwst/associations/lib/rules_level2b.py

```diff
@@ -354,6 +354,57 @@ def __init__(self, *args, **kwargs):
         self.constraints = Constraint([
             Constraint_Base(),
             Constraint_Single_Science(self.has_science),
+            DMSAttrConstraint(
+                name='opt_elem2',
+                sources=['grating'],
+                value='mirror'
+            ),
+            DMSAttrConstraint(
+                name='instrument',
+                sources=['instrume'],
+                value='nirspec'
+            ),
+            DMSAttrConstraint(
+                name='opt_elem',
+                sources=['filter'],
+                value='opaque'
+            ),
+        ])
+
+        super(Asn_Lv2NRSLAMPImage, self).__init__(*args, **kwargs)
+
+    def _init_hook(self, item):
+        """Post-check and pre-add initialization"""
+
+        super(Asn_Lv2NRSLAMPImage, self)._init_hook(item)
+        self.data['asn_type'] = 'nrslamp-image2'
+
+
+@RegistryMarker.rule
+class Asn_Lv2NRSLAMPSpectral(
+        AsnMixin_Lv2Special,
+        DMSLevel2bBase
+):
+    """Level2b NIRSpec spectral Lamp calibrations
+
+    NRS_LAMP exposures require specific level 2 processing.
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        self.constraints = Constraint([
+            Constraint_Base(),
+            Constraint_Single_Science(self.has_science),
+            Constraint(
+                [
+                    DMSAttrConstraint(
+                        name='opt_elem2',
+                        sources=['grating'],
+                        value='mirror'
```

Comment:
If this is the spectral form of NRS_LAMP, then why does it also have the grating='mirror' constraint (same as the image version)? The spectral form would have an actual grating.